packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -716,26 +736,64 @@ class _RangeSliderState extends State<RangeSlider> with TickerProviderStateMixin
+      cursor: effectiveMouseCursor,
       child: CompositedTransformTarget(
         link: _layerLink,
-        child: _RangeSliderRenderObjectWidget(
-          values: _unlerpRangeValues(widget.values),
-          divisions: widget.divisions,
-          labels: widget.labels,
-          sliderTheme: sliderTheme,
-          textScaleFactor: effectiveTextScale,
-          screenSize: screenSize(),
-          onChanged: _enabled && (widget.max > widget.min) ? _handleChanged : null,
-          onChangeStart: widget.onChangeStart != null ? _handleDragStart : null,
-          onChangeEnd: widget.onChangeEnd != null ? _handleDragEnd : null,
-          state: this,
-          semanticFormatterCallback: widget.semanticFormatterCallback,
-          hovering: _hovering,
+        child: FocusTraversalGroup(
+          policy: OrderedTraversalPolicy(),
+          child: _RangeSliderRenderObjectWidget(
+            key: _renderSliderKey,
+            values: _unlerpRangeValues(widget.values),
+            divisions: widget.divisions,
+            labels: widget.labels,
+            sliderTheme: sliderTheme,
+            textScaleFactor: effectiveTextScale,
+            screenSize: screenSize(),
+            onChanged: onChanged,
+            onChangeStart: widget.onChangeStart != null ? _handleDragStart : null,
+            onChangeEnd: widget.onChangeEnd != null ? _handleDragEnd : null,
+            state: this,
+            hovering: _hovering,
+            endThumb: FocusTraversalOrder(
+              order: const NumericFocusOrder(1),
+              child: _FocusableThumb(
+                renderThumbKey: _renderEndThumbKey,
+                values: _unlerpRangeValues(widget.values),
+                divisions: widget.divisions,
+                focusNode: _endThumbFocusNode,
+                sliderState: this,
+                thumb: Thumb.end,
+                sliderTheme: sliderTheme,
+                onChanged: onChanged,
+                renderSliderKey: _renderSliderKey,
+                targetPlatform: Theme.of(context).platform,
+                semanticFormatterCallback: widget.semanticFormatterCallback,
+              ),
+            ),
+            startThumb: FocusTraversalOrder(
+              order: const NumericFocusOrder(0),
+              child: _FocusableThumb(
+                values: _unlerpRangeValues(widget.values),
+                divisions: widget.divisions,
+                focusNode: _startThumbFocusNode,
+                sliderState: this,
+                thumb: Thumb.start,
+                sliderTheme: sliderTheme,
+                onChanged: onChanged,
+                renderSliderKey: _renderSliderKey,
+                targetPlatform: Theme.of(context).platform,
+                semanticFormatterCallback: widget.semanticFormatterCallback,
+                renderThumbKey: _renderStartThumbKey,
+              ),
+            ),
+            startThumbKey: _renderStartThumbKey,
```

Comment:
we should avoid using globalkey to set property on other render object directly, it may run into some weird lifecycle bug.

Consider use a changeNotifier to propagate information between renderobjects